crates/sui/tests/shared_objects_version_tests.rs

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::PathBuf;
+use sui_config::{NetworkConfig, ValidatorInfo};
+use sui_node::SuiNode;
+use sui_types::base_types::{ObjectRef, SequenceNumber};
+use sui_types::error::{SuiError, SuiResult};
+use sui_types::messages::{CallArg, ObjectArg, TransactionEffects};
+use sui_types::object::{Object, Owner, OBJECT_START_VERSION};
+use test_utils::authority::{spawn_test_authorities, test_authority_configs};
+use test_utils::messages::move_transaction;
+use test_utils::objects::test_gas_objects;
+use test_utils::transaction::{
+    publish_package, submit_shared_object_transaction, submit_single_owner_transaction,
+};
+
+#[tokio::test]
```

Comment:
i would highly recommend changing all tests in this file to `#[simtest]` and testing with `cargo simtest` - tests involving shared objects are notoriously flaky when using `#[tokio::test]`.